File: frontend/src/microservices/hotel/objects/BookHotelPlannerRequest.ts
export type BookHotelPlannerRequest = {
  userId: string
  roomTypeId: string
  guestTravelerIds: string[]
  checkInDate: string
  checkOutDate: string
  roomCount: number
}

export const bookHotelPlannerRequestFromJson = (json: string): BookHotelPlannerRequest =>
  JSON.parse(json) as BookHotelPlannerRequest

export const bookHotelPlannerRequestToJson = (value: BookHotelPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/hotel/objects/BookHotelPlannerRequest.scala
package com.typesafe.travel.hotel.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class BookHotelPlannerRequest(
    userId: String,
    roomTypeId: String,
    guestTravelerIds: List[String],
    checkInDate: String,
    checkOutDate: String,
    roomCount: Int
)

object BookHotelPlannerRequest:
  given sourceEncoder: Encoder[BookHotelPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[BookHotelPlannerRequest] = deriveDecoder
